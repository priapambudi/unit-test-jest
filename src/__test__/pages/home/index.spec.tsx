import React from "react";
import { fireEvent, render } from "@testing-library/react";
import HomepageView from "../../../components/pages/Home";

describe("render home", () => {
  it("should render home", () => {
    const { container } = render(<HomepageView />);
    expect(container).toMatchSnapshot();
  });
  it("should click button submit on home page", () => {
    const { getByText } = render(<HomepageView />);
    const buttonSubmit = getByText("Submit");
    fireEvent.click(buttonSubmit);
    expect(buttonSubmit).toBeInTheDocument();
  });
});
