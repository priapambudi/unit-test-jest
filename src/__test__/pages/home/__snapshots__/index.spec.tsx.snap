// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home should render home 1`] = `
<div>
  <div>
    <h1>
      Add a User
    </h1>
    <p>
      Form
    </p>
    <label
      for="name"
    >
      name
    </label>
     
    <br />
    <input
      id="name"
      type="text"
    />
    <br />
    <label
      for="email"
    >
      email
    </label>
     
    <br />
    <input
      id="email"
      type="text"
    />
    <br />
    <button
      class="primary"
    >
      Submit
    </button>
  </div>
</div>
`;
